### PR TITLE
Removed MetadataReferenceFeature from the Application Parts sample.

### LINKED ISSUE #12316: MetadataReferenceFeature is obsolete: (https://aka.ms/AA1x4gg)
Hi team,
I'm moving this from here: https://github.com/aspnet/AspNetCore.Docs.de-de/issues/38
The user is reporting obsolete content in the code box.

@cyberh0me commented:
please could you update the FeaturesController sample accordingly

See https://github.com/aspnet/Announcements/issues/312

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2784a724-e940-4c14-3b50-b063f78c2ee5
* Version Independent ID: 41175c3a-8cb5-856b-a7e0-931edb6394db
* Content: [Application Parts in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/advanced/app-parts?view=aspnetcore-2.2#feedback)
* Content Source: [aspnetcore/mvc/advanced/app-parts.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/advanced/app-parts.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #12316 

The samples need updating (they're currently at 2.0), but I've left that out of this to keep things focused on the issue at hand.

### CODE REVIEW

File: aspnetcore/mvc/advanced/app-parts.md [hunk 1]
```diff
@@ -91,16 +90,39 @@ The `GenericController` class:
 
 [!code-csharp[](./app-parts/sample/AppPartsSample/GenericController.cs?highlight=5-6)]
 
-The result, when a matching route is requested:
+The result when requesting `http://localhost:28431/Sprocket`:
```
Comment (serpent5) [inline aspnetcore/mvc/advanced/app-parts.md]: As we're removing the other screenshot, it makes sense to remove this one. It showed the URL, which is important to the example, so I included that in the text.

File: aspnetcore/mvc/advanced/app-parts.md [hunk 2]
```diff
@@ -91,16 +90,39 @@ The `GenericController` class:
...
 
-![Example output from the sample app reads, 'Hello from a generic Sprocket controller.'](app-parts/_static/generic-controller.png)
+```text
+Hello from a generic Sprocket controller.
```
Comment (serpent5) [inline aspnetcore/mvc/advanced/app-parts.md]: The output was shown in a screenshot, so I've pulled it in as text instead. I dunno if this is really needed, but I didn't want to _just_ remove the image.

File: aspnetcore/mvc/advanced/app-parts.md [hunk 3]
```diff
@@ -91,16 +90,39 @@ The `GenericController` class:
...
 You can iterate through the populated features available to your app by requesting an `ApplicationPartManager` through [dependency injection](../../fundamentals/dependency-injection.md) and using it to populate instances of the appropriate features:
 
-[!code-csharp[](./app-parts/sample/AppPartsSample/Controllers/FeaturesController.cs?highlight=16,25-27)]
+[!code-csharp[](./app-parts/sample/AppPartsSample/Controllers/FeaturesController.cs?highlight=15,24-26)]
 
 Example output:
 
-![Example output from the sample app](app-parts/_static/available-features.png)
+```text
+Controllers:
```
Comment (serpent5) [inline aspnetcore/mvc/advanced/app-parts.md]: Same here - removed the image and replaced it with the text output.

File: aspnetcore/mvc/advanced/app-parts.md [hunk 4]
```diff
@@ -91,16 +90,39 @@ The `GenericController` class:
...
+
+Tag Helpers:
+    - AnchorTagHelper
+    - CacheTagHelper
+    - DistributedCacheTagHelper
+    - EnvironmentTagHelper
+    - FormTagHelper
+
+View Components:
+    - AnchorTagHelper
```
Comment (serpent5) [inline aspnetcore/mvc/advanced/app-parts.md]: Yeah, these are the same as the tag-helpers section. It's a bad copy/paste in the view file, which I suggest we fix when updating the samples to 3.0.

**Comment by Rick-Anderson** [comment]
@serpent5  nice work, thanks!

**Pull request merged**